kitsune/kpi/tests/test_cron.py

```diff
@@ -1,24 +1,153 @@
-from datetime import date, timedelta
+from datetime import date, datetime, timedelta
 
 from mock import patch
 from nose.tools import eq_
 
 import kitsune.kpi.cron
+from kitsune.customercare.tests import ReplyFactory
 from kitsune.kpi import surveygizmo_utils
 from kitsune.kpi.cron import (
-    update_visitors_metric, update_l10n_metric, googleanalytics,
+    cohort_analysis, update_visitors_metric, update_l10n_metric, googleanalytics,
     update_search_ctr_metric, _process_exit_survey_results)
 from kitsune.kpi.models import (
-    Metric, VISITORS_METRIC_CODE, L10N_METRIC_CODE, SEARCH_CLICKS_METRIC_CODE,
+    Metric, Cohort, VISITORS_METRIC_CODE, L10N_METRIC_CODE, SEARCH_CLICKS_METRIC_CODE,
     SEARCH_SEARCHES_METRIC_CODE, EXIT_SURVEY_YES_CODE, EXIT_SURVEY_NO_CODE,
-    EXIT_SURVEY_DONT_KNOW_CODE)
+    EXIT_SURVEY_DONT_KNOW_CODE, CONTRIBUTOR_COHORT_CODE, KB_CONTRIBUTOR_COHORT_CODE,
+    KB_L10N_CONTRIBUTOR_COHORT_CODE, SUPPORT_FORUM_HELPER_COHORT_CODE, AOA_CONTRIBUTOR_COHORT_CODE)
 from kitsune.kpi.tests import MetricKindFactory, MetricFactory
+from kitsune.questions.tests import AnswerFactory
 from kitsune.sumo.tests import TestCase
+from kitsune.users.tests import ProfileFactory
 from kitsune.wiki.config import (
     MAJOR_SIGNIFICANCE, MEDIUM_SIGNIFICANCE, TYPO_SIGNIFICANCE)
 from kitsune.wiki.tests import DocumentFactory, ApprovedRevisionFactory
 
 
+class CohortAnalysisTests(TestCase):
+    def setUp(self):
+        today = datetime.today()
+        self.start_of_first_week = today - timedelta(days=today.weekday(), weeks=12)
+
+        revisions = ApprovedRevisionFactory.create_batch(3, created=self.start_of_first_week)
+
+        reviewer = ProfileFactory()
```

Comment:
Oh yeah! I'm so used to the old create a profile, then use the user thing we used to have to do.